backend/src/main/java/gov/cdc/usds/simplereport/service/LoincService.java

```diff
@@ -0,0 +1,270 @@
+import feign.Response;
+import gov.cdc.usds.simplereport.config.AuthorizationConfiguration;
+import gov.cdc.usds.simplereport.db.model.Condition;
+import gov.cdc.usds.simplereport.db.model.Lab;
+import gov.cdc.usds.simplereport.db.model.LoincStaging;
+import gov.cdc.usds.simplereport.db.repository.LabRepository;
+import gov.cdc.usds.simplereport.db.repository.LoincStagingRepository;
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.concurrent.CompletableFuture;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.commons.io.IOUtils;
+import org.apache.http.HttpStatus;
+import org.hl7.fhir.r4.model.Coding;
+import org.hl7.fhir.r4.model.Parameters;
+import org.hl7.fhir.r4.model.Type;
+import org.springframework.data.domain.Page;
+import org.springframework.data.domain.PageRequest;
+import org.springframework.scheduling.annotation.Async;
+import org.springframework.stereotype.Service;
+
+@Service
+@RequiredArgsConstructor
+@Slf4j
+public class LoincService {
+
+  private final LoincFhirClient loincFhirClient;
+  private final LoincStagingRepository loincStagingRepository;
+  private final LabRepository labRepository;
+  private final FhirContext context = FhirContext.forR4();
+  private IParser parser = context.newJsonParser();
+  private static final int PAGE_SIZE = 20;
+
+  @AuthorizationConfiguration.RequireGlobalAdminUser
+  @Async
+  public void syncLabs() {
+    log.info("Sync Labs");
+    PageRequest pageRequest = PageRequest.of(0, PAGE_SIZE);
+    List<CompletableFuture<Response>> futures = new ArrayList<>();
+    List<Lab> labs = new ArrayList<>();
+    List<LoincStaging> failedLoincs = new ArrayList<>();
+    List<LoincStaging> successLoincs = new ArrayList<>();
+    Page<LoincStaging> loincPage = loincStagingRepository.findAll(pageRequest);
+
+    while (loincPage.hasNext()) {
+      List<LoincStaging> loincs = loincPage.getContent();
+      log.info("Found {} Labs", loincs.size());
+      loincs.forEach(
+          loinc ->
+              futures.add(
+                  CompletableFuture.supplyAsync(
+                      () -> loincFhirClient.getCodeSystemLookup(loinc.getCode()))));
+      log.info("Futures created");
+      CompletableFuture<Void> allFutures =
+          CompletableFuture.allOf(futures.toArray(new CompletableFuture[0]));
+      allFutures.join();
+      log.info("Futures completed");
+      for (int i = 0; i < futures.size(); i++) {
+        Response response = futures.get(i).getNow(null);
+        LoincStaging loinc = loincs.get(i);
+        if (response.status() != HttpStatus.SC_OK) {
+          failedLoincs.add(loinc);
+          log.error(
+              "Received a {} status code from the LOINC API response for: {}",
+              response.status(),
+              loinc.getCode());
+          continue;
```

Comment:
I think this is probably true for all of the external API usages, but we should probably consider accounting for service disruptions and short circuiting these syncs when appropriate.